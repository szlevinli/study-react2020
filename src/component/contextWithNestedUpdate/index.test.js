import { render } from '@testing-library/react';
import React from 'react';
import ThemedTogglerButton from './index';
import userEvent from '@testing-library/user-event';

it('should be render ThemedTogglerButton correctly', () => {
  const { container } = render(<ThemedTogglerButton />);
  expect(container).toMatchSnapshot();
});

it('should be toggle theme', () => {
  const { getByRole } = render(<ThemedTogglerButton />);
  const button = getByRole('button');
  expect(button).toHaveAttribute('style', expect.stringContaining('238'));
  userEvent.click(button);
  expect(button).toHaveAttribute('style', expect.stringContaining('34'));
});
