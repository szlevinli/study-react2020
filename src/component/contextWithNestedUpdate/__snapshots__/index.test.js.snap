// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be render ThemedTogglerButton correctly 1`] = `
<div>
  <article>
    <h1>
      Context
    </h1>
    <button
      style="background-color: rgb(238, 238, 238);"
    >
      Toggle Theme
    </button>
  </article>
</div>
`;
